core/src/main/java/org/apache/calcite/sql2rel/RelDecorrelator.java

```diff
@@ -1032,6 +1033,13 @@ public Frame decorrelateRel(LogicalSnapshot rel) {
     return decorrelateRel((RelNode) rel);
   }
 
+  public Frame decorrelateRel(LogicalTableFunctionScan rel) {
```

Comment:
This class also contains methods for non-logical rel nodes, for example, `decorrelateRel(Filter rel)`. Should be `TableFunctionScan` there instead of `LogicalTableFunctionScan` to ensure that this issue wouldn't be reproduced for other implementations of `TableFunctionScan`?

Reply:
Decoration should only happen for logical nodes, we keep node Filter only for compatibility with old version Calcite.

Reply:
Ok, thanks for the explanation.